lightning/src/ln/onion_utils.rs

```diff
@@ -446,10 +449,14 @@ pub(super) fn process_onion_failure<T: secp256k1::Signing, L: Deref>(secp_ctx: &
 							});
 						}
 
+						if short_channel_id.is_none() {
+							short_channel_id = Some(route_hop.short_channel_id);
```

Comment:
Shouldn't this be the next-hop not the current hop, at least in the case of a `!is_chan_update_invalid` ChannelUpdate in the `UPDATE` block?

Reply:
Are you suggesting setting it to what the `ChannelUpdate` contains in that case such that this code path is not triggered?

Reply:
Basically, yes, though we should check the ChannelUpdate contains the expected SCID first.

Reply:
Ok, I think I've now implemented as described. Please take a look at the fixup commit and let me know if it looks sane.